### PR TITLE
WIP: Django 1.9 updates and fixes

### PR DESCRIPTION
**Pull request by mariocesar**
Remove `django.conf.urls.patterns`, since they already have been removed from the doc and will be removed in Django 2.0

This removes in Django 1.9 the warning:
```
RemovedInDjango110Warning: django.conf.urls.patterns()
is deprecated and will be removed in Django 1.10. Update your urlpatterns to be a list of
django.conf.urls.url() instances instead.
  name='redactor_upload_file'),
```

Also Updates the command `update_index` to match new arguments parser in Django

### CODE REVIEW

**Comment by acdha** [comment]
This is looking promising – the last failure is probably just a minor change to related to the command-line processing cleanup:

https://travis-ci.org/django-haystack/django-haystack/jobs/93071939#L1408-L1424

**Comment by gustavi** [comment]
Some news of this PR @mariocesar  ?

**Comment by mariocesar** [comment]
@gustavi I neglect this branch for some weeks, Sorry :(

It needs some minor refactor to make it work, Will try to make it the next week.

If you want to help, there are renaming and fixing imports issues, there are not big problems. Just make a PR for my patch branch

**Comment by joshowen** [comment]
Thanks for working on this @mariocesar!

**Comment by gustavi** [comment]
Sorry @mariocesar but I don"t know the code. But this is really a bloquer for our project.

**Comment by gustavi** [comment]
No news ?

File: haystack/fields.py [hunk 1]
```diff
@@ -14,7 +15,26 @@ class NOT_PROVIDED:
     pass
 
 
-DATETIME_REGEX = re.compile('^(?P<year>\d{4})-(?P<month>\d{2})-(?P<day>\d{2})(T|\s+)(?P<hour>\d{2}):(?P<minute>\d{2}):(?P<second>\d{2}).*?$')
+DATETIME_REGEX = re.compile('''
+^
+(?P<year>\d{4})-
+(?P<month>\d{2})-
+(?P<day>\d{2})
+(T|\s+)              # Time flag or an space
```
Comment (gustavi) [inline haystack/fields.py]: not sure this comment is correct

**Comment by mariocesar** [comment]
I'm back!

File: haystack/fields.py (hunk 1 above)
Comment (mariocesar) [inline haystack/fields.py]: Did you mean \s+ ? Could be better, one space chars or more

**Comment by sk1p** [comment]
As I already commented on #1300, I fixed the last test failures in my fork: https://github.com/sk1p/django-haystack/tree/django_19, feel free to merge these changes into your pull request (might need the changes from #1300, too).

**Comment by acdha** [comment]
Sorry about the delays but as of 10bcc7668ccaab499f8db516fd59de182aadbcb4 the tests are passing on Django 1.9. If you have any remaining gaps please either rebase against master or close this & open a separate pull request.

**Comment by Gesias** [comment]
Thanks for your efforts!

**Pull request closed without merging**